BuildScripts/lib/BuildCore.ps1

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env pwsh
+#Requires -Version 5.0
+
+# Imports
+. $PSScriptRoot\Constants.ps1
+. $PSScriptRoot\Enums.ps1
+. $PSScriptRoot\FileTools.ps1
+. $PSScriptRoot\MoveTools.ps1
+. $PSScriptRoot\TFMTools.ps1
+. $PSScriptRoot\EngineMap.ps1
+
+function Build-Project {
+  [CmdletBinding()]
+  param (
+    [Parameter(Mandatory)]
+    [Games]
+    $Game,
+    [Parameter(Mandatory)]
+    [Engines]
+    $Engine,
+    [Parameter(Mandatory)]
+    [Configurations]
+    $Configuration,
+    [Parameter(Mandatory)]
+    [string]
+    $TargetFramework,
+    [Parameter()]
+    [Switch]
+    [bool]
+    $SkipMoveCommonLibraries
+  )
+
+  begin {
+    $Private:TargetFrameworkWithoutTFM = Get-TargetFrameworkWithoutTFM $TargetFramework
+    $Private:SpecialName = Get-PlatformName $Engine
+    $Private:ClientSuffix = Get-Suffix $Engine
+
+    $Private:DotnetArgs = @(
+      "build"
```

Comment:
Please use dotnet publish instead of dotnet build. dotnet publish is the only official way to make a 'production ready' release.
![image](https://user-images.githubusercontent.com/25006126/174470995-f7156763-b714-4dbd-86bf-579e1316bbe0.png)